### DISCUSSION TITLE
Conan integration, how to set 'needs_exe_wrapper'

### DISCUSSION THREAD

**Discussion by avetiso** [mesonbuild/meson#10127] [Q&A] [answered]
Hi,

I'm trying to set up cross compilation solely via conan and the cross files generated by the conan meson toolchain.

Running into the well-known issue of the sanity check running on iOS binaries and failing. I wish I could add `needs_exe_wrapper` to [properties], but this file is generated and conan currently does not expose any way for me to set any fields in [properties].

I've opened an MR for them here: https://github.com/conan-io/conan/issues/10796

But in the meantime, what would be the idiomatic way of detecting this case in meson.build and deactivating the sanity check?

I've seen functions for getting properties, but not for setting them. In fact I suspect that once the cross file is consumed there's no way of changing anything after it.

**Comment by eli-schwartz** [comment, marked as answer]
This would violate Meson's design rationale, that projects should not encode information about the build environment into `meson.build`. It's the same reason you cannot define the C/C++ compiler in `meson.build`.

What you can do, if conan allows it, is pass a second cross file, this one containing your user specified properties. Meson allows layering cross files such that some settings come from one file and other settings come from another file.

And of course, conan can fix the bug, and Meson's design rationale to force it to be fixed in the cross file generation by conan, means that projects won't end up with conan specific workarounds for conan bugs inside their project definitions. :)

**Reply by avetiso** [reply]
Awesome, I'll go ahead and give layering the cross files a try and see if it works in the meantime. However, it does sound like this definitely on conan to fix.

Thank you!
